pkg/common/plugin/devid/devid.go

```diff
@@ -0,0 +1,48 @@
+package devid
+
+import (
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const PluginName = "devid"
+
+type AttestationRequest struct {
+	DevIDCert []byte
+	DevIDPub  []byte
+
+	EKCert []byte
+	EKPub  []byte
```

Comment:
Why do we have to send the server public keys if we're also sending it certs, which presumably have the same public keys in them?

Reply:
We send both to verify that the public key in the certificate matches the key regenerated by the TPM.

The Endorsement Certificate is provided by the manufacturer and stored in a predefined NV index in the TPM. However, the Endorsement Key object that contains the public and private keys needs to be regenerated based on a  TPM seed and a key template.

We need the **Endorsement Key object** (which contains the public key and private key) loaded into the TPM to do the credential activation (verify that another key is in the same TPM as the EK). The server uses this public key to generate the credential activation challenge.

We need the **Endorsement certificate** to establish a chain of trust up to one of the endorsement roots certificates.

It is important to verify that both keys match to "join" these two affirmations.